aws/resource_aws_launch_template.go

```diff
@@ -797,19 +805,56 @@ func resourceAwsLaunchTemplateUpdate(d *schema.ResourceData, meta interface{}) e
 		return err
 	}
 
-	launchTemplateVersionOpts := &ec2.CreateLaunchTemplateVersionInput{
-		ClientToken:        aws.String(resource.UniqueId()),
-		LaunchTemplateId:   aws.String(d.Id()),
-		LaunchTemplateData: launchTemplateData,
+	dltv, err := conn.DescribeLaunchTemplateVersions(&ec2.DescribeLaunchTemplateVersionsInput{
+		LaunchTemplateId: aws.String(d.Id()),
+		Versions:         aws.StringSlice([]string{"$Latest"}),
+	})
+	if err != nil {
+		return fmt.Errorf("Error describing $Latest Launch Template Version for Launch Template(%s): %w", d.Id(), err)
+	}
+	if dltv == nil || len(dltv.LaunchTemplateVersions) == 0 {
+		return fmt.Errorf("Error describing $Latest Launch Template Version for Launch Template(%s): empty output", d.Id())
 	}
 
-	if v, ok := d.GetOk("description"); ok && v.(string) != "" {
-		launchTemplateVersionOpts.VersionDescription = aws.String(v.(string))
+	// Create new version only if content has changed since previous version
+	var defaultVersion string
+	description := d.Get("description").(string)
+	latestVersionDescription := aws.StringValue(dltv.LaunchTemplateVersions[0].VersionDescription)
+
+	if (launchTemplateData.String() != dltv.LaunchTemplateVersions[0].LaunchTemplateData.String()) || (description != latestVersionDescription) {
```

Comment:
While it will be a little unwieldy until something like https://github.com/hashicorp/terraform-plugin-sdk/issues/457 is merged, I think rather than depending like logic on this (which potentially can be very brittle based on whitespace changes, nil vs zero-values, etc.) or spending the time to write a large `(*ec2.LaunchTemplateData).Equals()`-like function to appropriately check the equality of these two items, we should consider using a `d.HasChanges()` conditional for all the fields except `tags`/`update_default_version`. This should allow the original update logic to exist without the equality checking or additional API call. e.g.

```go
	conn := meta.(*AWSClient).ec2conn

	version := d.Get("latest_version").(int)

	if d.HasChanges(
		"block_device_mappings",
		"capacity_reservation_specification",
		"cpu_options",
		"credit_specification",
		"description",
		"disable_api_termination",
		"elastic_gpu_specifications",
		"elastic_inference_accelerator",
		"hiberation_options",
		"iam_instance_profile",
		"image_id",
		"instance_initiated_shutdown_behavior",
		"instance_market_options",
		"instance_type",
		"kernel_id",
		"key_name",
		"license_specification",
		"metadata_options",
		"monitoring",
		"name",
		"name_prefix",
		"network_interfaces",
		"placement",
		"ram_disk_id",
		"security_group_names",
		"tag_specifications",
		"user_data",
		"vpc_security_group_ids",
	) {
		launchTemplateData, err := buildLaunchTemplateData(d)

		if err != nil {
			return err
		}

		input := &ec2.CreateLaunchTemplateVersionInput{
			ClientToken:        aws.String(resource.UniqueId()),
			LaunchTemplateId:   aws.String(d.Id()),
			LaunchTemplateData: launchTemplateData,
		}

		if v, ok := d.GetOk("description"); ok {
			input.VersionDescription = aws.String(v.(string))
		}

		output, err := conn.CreateLaunchTemplateVersion(input)

		if err != nil {
			return fmt.Errorf("error creating Launch Template (%s) Version: %w", d.Id(), err)
		}

		if output != nil && output.LaunchTemplateVersion != nil && output.LaunchTemplateVersion.VersionNumber != nil {
			version = aws.Int64Value(output.LaunchTemplateVersion.VersionNumber)
		}
	}

	if d.Get("update_default_version").(bool) || (d.HasChange("default_version") && d.Get("default_version").(int) > 0) {
		input := &ec2.ModifyLaunchTemplateInput{
			LaunchTemplateId: aws.String(d.Id()),
			DefaultVersion: aws.String(strconv.FormatInt(version, 10)),
		}

		_, err = conn.ModifyLaunchTemplate(input)

		if err != nil {
			return fmt.Errorf("error modifying Launch Template (%s): %w", d.Id(), err)
		}
	}

	if d.HasChanges("tags") {
// ...
```

Reply:
@bflad, only note is that "name" and "name-prefix" are `ForceNew` attributes, so I've excluded them from the list of keys with changes to look for